experimental/sql/dialects/rel_ssa.py

```diff
@@ -304,6 +304,51 @@ def get(input: Operation, predicates: Region) -> 'Select':
                         result_types=[input.results[0].typ])
 
 
+@irdl_op_definition
+class Aggregate(Operator):
+  """
+  Applies the ith function of `functions` to the ith column name fo `col_names`
+  of `input`.
+
+
+  Example:
+
+  '''
+  %0 : !rel_ssa.bag<[!rel_ssa.schema_element<"a0", !rel_ssa.int32>]> = rel_ssa.aggregate(%0 : !rel_ssa.bag<[!rel_ssa.schema_element<"id", !rel_ssa.int32>]>) ["col_names" = ["id"], "functions" = ["sum"]]
```

Comment:
I do not understand this modeling. 
Where is `a0` coming from in the result type `!rel_ssa.bag<[!rel_ssa.schema_element<"a0", !rel_ssa.int32>]>` ?

Also, this performs a reduction but the types for the input and output seem to be the same which feels fishy.

Reply:
Yes the types do seem fishy in this case, but it actually happens to be correct here since the sum reduction results in another integer.

The a0 hints at a problem that we have with reductions: Since they create a column, we have to name that column. But the name should be unique, so we need to be careful. The current version just gives the same name as the input column since that column won't be part of the result anymore. It seems that I forgot to update this in the documentation.